pages/applications/view/[id]/index.tsx

```diff
@@ -56,15 +57,32 @@ export default function ApplicationPage({
 }: PageProps): JSX.Element | null {
   if (!data.id) return <Custom404 />;
 
-  // Can edit application if:
-  // - it has a status of manual draft
-  // - it has a status of incomplete and current user is assigned to it
-  const canEditApplication =
-    data.status === ApplicationStatus.MANUAL_DRAFT ||
-    (data.status === 'New' && data.assignedTo === user.email);
-
   const [activeNavItem, setActiveNavItem] = useState('overview');
 
+  // Can edit applications if:
+  // - user is a manager (all statuses)
+  // - it has a status of manual draft
+  // - it has a status of awaiting assessment (SUBMITTED) and current user is assigned to it
+  // - it has a status of awaiting reassessment and current user is assigned to it
+  const canEditApplications = () => {
+    if (!hasAnyPermissions(user)) return false;
+    if (user.hasManagerPermissions) return true;
+    if (data.status === ApplicationStatus.MANUAL_DRAFT) {
+      return true;
+    }
+    const assignedToCurrentUser = data.assignedTo === user.email;
+    if (data.status === ApplicationStatus.SUBMITTED && assignedToCurrentUser) {
+      return true;
+    }
+    if (
+      data.status === ApplicationStatus.AWAITING_REASSESSMENT &&
+      assignedToCurrentUser
+    ) {
+      return true;
+    }
+    return false;
+  };
```

Comment:
This is great, but I do think you might need to replicate some similar logic at the API level to ensure that these permissions are properly enforceable

Reply:
Maybe putting this and other admin permissions logic at the API level should be in a future ticket?

The reason being that similar tests already exist such as `canViewSensitiveApplication` in `googleAuth.ts` which use this convention (checking the properties on `user`).